src/scene/scene.js

```diff
@@ -176,6 +183,108 @@ class Scene extends EventHandler {
         this.off();
     }
 
+    get clusteredLightingEnabled() {
+        return this._clusteredLightingEnabled;
+    }
+
+    set clusteredLightingEnabled(value) {
+
+        if (this._clusteredLightingEnabled && !value) {
+            console.error("Turning off enabled clustered lighting is not currently supported");
+            return;
+        }
+
+        this._clusteredLightingEnabled = value;
+    }
+
+    get lightCells() {
+        return this.worldClustersParams.cells;
+    }
+
+    set lightCells(value) {
+        this.worldClustersParams.cells.copy(value);
+    }
+
+    get maxLights() {
+        return this.worldClustersParams.maxLights;
+    }
+
+    set maxLights(value) {
+        this.worldClustersParams.maxLights = value;
+    }
+
+    get cookieAtlasResolution() {
+        return this.worldClustersParams.cookieAtlasResolution;
+    }
+
+    set cookieAtlasResolution(value) {
+        this.worldClustersParams.cookieAtlasResolution = value;
+    }
+
+    get shadowAtlasResolution() {
+        return this.worldClustersParams.shadowAtlasResolution;
+    }
+
+    set shadowAtlasResolution(value) {
+        this.worldClustersParams.shadowAtlasResolution = value;
+    }
+
+    get shadowType() {
+        return this.worldClustersParams.shadowType;
+    }
+
+    set shadowType(value) {
+        if (this.worldClustersParams.shadowType !== value) {
+            this.worldClustersParams.shadowType = value;
+
+            // lit shaders need to be rebuilt
+            this._layers._dirtyLights = true;
+        }
+    }
+
+    get cookiesEnabled() {
+        return this.worldClustersParams.cookiesEnabled;
+    }
+
+    set cookiesEnabled(value) {
+        if (this.worldClustersParams.cookiesEnabled !== value) {
+            this.worldClustersParams.cookiesEnabled = value;
+
+            // lit shaders need to be rebuilt
+            this._layers._dirtyLights = true;
+        }
+    }
+
+    get areaLightsEnabled() {
+        return this.worldClustersParams.areaLightsEnabled;
+    }
+
+    set areaLightsEnabled(value) {
+
+        // ignore if not supported
+        if (this.device.supportsAreaLights) {
```

Comment:
Seems a waste to store device on the scene just for this. Could whoever uses `supportsAreaLights` flag also perform the device check?

Reply:
What is the problem with storing the Device there?

I don't like to move the responsibility of this to the user. They want area lights. It's our problem we don't support them on all devices and need to block it on some? Similarly to for example PCF5 - user requests it, we use PCF3 if not supported. And many other examples.